CloudFlareUtilities/ChallengeSolution.cs

```diff
@@ -22,10 +23,14 @@ public ChallengeSolution(string clearancePage, string verificationCode, string p
 
         public string Pass { get; }
 
+        public string S { get; }
+
         public double Answer { get; }
 
         // Using .ToString("R") to reduse answer rounding
-        public string ClearanceQuery => $"{ClearancePage}?jschl_vc={VerificationCode}&pass={Pass}&jschl_answer={Answer.ToString("R", CultureInfo.InvariantCulture)}";
+        public string ClearanceQuery => !(S is null) ? 
```

Comment:
@Bleialf => it would be better to write `!(string.IsNullOrEmpty(S))` instead of `!(S is null)`

S could be an empty string (but not null) due to 
`var s = Regex.Match(challengePageContent, "name=\"s\" value=\"(?<s>[^\"]+)").Groups["s"].Value;`
You can verify it with challengePageContent being a string without s parameter

Reply:
You are absolutely right @Eric898989